transit/script/bart.py

```diff
@@ -0,0 +1,396 @@
+
+    # Mapping from Node and Station DataFrames
+    node_mapping = nodes.set_index("Node")["Node Name"].to_dict()
+    station_mapping = df_station_name.set_index("Node")["Station"].to_dict()
+
+    # Apply mappings
+    intra["A_name"] = intra["A"].map(node_mapping)
+    intra["B_name"] = intra["B"].map(node_mapping)
+    intra["A_station"] = intra["A"].map(station_mapping)
+    intra["B_station"] = intra["B"].map(station_mapping)
+
+    # Filter out specific values and stations
+    values_to_drop = ["Hillcrest eBART", "Coliseium OAC", "Somersville Road eBART"]
+    intra = intra[
+        ~intra["A_name"].isin(values_to_drop) & ~intra["B_name"].isin(values_to_drop)
+    ]
+    intra = intra[
+        intra["A_station"].isin(relevant_stations)
+        & intra["B_station"].isin(relevant_stations)
+    ]
+
+    # Apply custom station labeling and direction determination
+    intra["A_SL"] = intra["A_station"].apply(lambda x: station_to_label[x])
+    intra["B_SL"] = intra["B_station"].apply(lambda x: station_to_label[x])
+    intra["Direction"] = intra.apply(lambda x: determineSL(x), axis=1)
+    intra = intra[intra["Direction"] != False]
+
+    # Set time of day and select final columns
+    intra = intra[["Direction", "TOD", "AB_VOL"]]
+    # 2019 validation uses 'Intra-SF'
+    intra["Screenline"] = "SF-San Mateo"
+    intra["Key"] = intra["Screenline"] + intra["Direction"] + intra["TOD"]
+    intra.columns = ["Direction", "TOD", "Ridership", "Screenline", "Key"]
+    intra = intra[["Screenline", "Direction", "TOD", "Key", "Ridership"]]
+    intra = intra.sort_values(by=["Direction", "TOD"]).reset_index(drop=True)
+
+    return intra
+
+
+def process_BART_SL(file_name, model_run_dir, output_dir, model_BART_SL):
+    transbay_node = [16510, 16511]  # 16510 in, 16511 out
+    BART_sl_tb = BART_SL_Concat(
+        file_name, transbay_node[0], transbay_node[1], "Transbay"
+    )
+    countyline_node = [16519, 16518]  # 16519n-- in, 16518 --out
+    BART_sl_ct = BART_SL_Concat(
+        file_name, countyline_node[0], countyline_node[1], "Countyline"
+    )
+    BART_sf = process_BART_SF(file_name, model_run_dir)
+    BART_SL = pd.concat([BART_sl_tb, BART_sl_ct, BART_sf], ignore_index=True)
+    BART_SL.to_csv(output_dir / model_BART_SL, index=False)
+
+
+if __name__ == "__main__":
+    with open("transit.toml", "rb") as f:
+        config = tomllib.load(f)
+
+    model_run_dir = config["directories"]["model_run"]
+    model_BART = config["output"]["model_BART"]
+    model_BART_county = config["output"]["model_BART_county"]
+    model_BART_SL = config["output"]["model_BART_SL"]
+    output_dir = model_run_dir / "validation_workbook" / "output"
+    output_transit_dir = output_dir / "transit"
+    output_transit_dir.mkdir(parents=True, exist_ok=True)
+    time_periods = ["EA", "AM", "MD", "PM", "EV"]
+    dbf_file = read_transit_assignments(model_run_dir, time_periods)
+
+    process_BART_SL(dbf_file, model_run_dir, output_dir, model_BART_SL)
```

Comment:
you could even have a process_BART() function that encapsulates line 392-396 (or 384-396 if you don't want to specify all the variables)